mm2src/coins/tendermint/tendermint_tx_history_v2.rs

```diff
@@ -0,0 +1,803 @@
+                                .value
+                                .to_string(),
+                        );
+                    },
+                    COIN_RECEIVED_EVENT if to.is_none() => {
+                        to = Some(
+                            event
+                                .attributes
+                                .iter()
+                                .find(|tag| tag.key.to_string() == RECEIVER_TAG_KEY)?
+                                .value
+                                .to_string(),
+                        );
+                    },
+                    _ => {},
+                }
+            }
+
+            Some(TransferDetails {
+                from: from?,
+                to: to?,
+                denom: denom?,
+                amount: amount?,
+            })
+        }
+
+        fn get_transfer_details(tx_events: Vec<Event>, fee_amount_with_denom: String) -> Option<TransferDetails> {
+            // Filter out irrelevant events
+            let mut events: Vec<&Event> = tx_events
+                .iter()
+                .filter(|event| [COIN_SPENT_EVENT, COIN_RECEIVED_EVENT].contains(&event.type_str.as_str()))
+                .collect();
+
+            if events.len() > DEFAULT_TRANSFER_EVENT_COUNT {
+                // Retain fee related events
+                events.retain(|event| {
+                    let amount_with_denom = event
+                        .attributes
+                        .iter()
+                        .find(|tag| tag.key.to_string() == AMOUNT_TAG_KEY)
+                        .map(|t| t.value.to_string());
+
+                    amount_with_denom != Some(fee_amount_with_denom.clone())
+                });
+            }
+
+            parse_transfer_values_from_events(events)
+        }
+
+        async fn fetch_and_insert_txs<Coin, Storage>(
+            tendermint_assets_conf: &[Json],
+            address: String,
+            coin: &Coin,
+            storage: &Storage,
+            query: String,
+            from_height: u64,
+        ) -> Result<u64, Stopped<Coin, Storage>>
+        where
+            Coin: TendermintTxHistoryOps,
+            Storage: TxHistoryStorage,
+        {
+            let mut page = 1;
+            let mut iterate_more = true;
+            let mut highest_height = from_height;
+
+            let client = try_or_return_stopped_as_err!(
+                coin.get_rpc_client().await,
+                StopReason::RpcClient,
+                "could not get rpc client"
+            );
+
+            while iterate_more {
+                let response = try_or_return_stopped_as_err!(
+                    client
+                        .perform(TxSearchRequest::new(
+                            query.clone(),
+                            false,
+                            page,
+                            TX_PAGE_SIZE,
+                            TendermintResultOrder::Ascending.into(),
```

Comment:
A question: do we first fetch most recent or oldest transactions?

Reply:
from oldest to latest